cnd/src/http_api/routes/index.rs

```diff
@@ -83,16 +84,24 @@ pub async fn post_han_ethereum_halight_bitcoin(
         .map_err(problem::from_anyhow)
         .map_err(warp::reject::custom)?;
 
-    let reply = warp::reply::reply();
+    let swap_params: HanEtherereumHalightBitcoinCreateSwapParams = body.clone().into();
 
-    let id = LocalSwapId::default();
+    let swap_id = LocalSwapId::default();
+    let reply = warp::reply::reply();
 
-    facade.save(id, ()).await;
+    let swap: CreatedSwap<HanEthereumEther, HalightLightningBitcoin> = CreatedSwap {
+        swap_id,
+        alpha: body.alpha,
+        beta: body.beta,
+        peer: body.peer,
+        role: body.role.0,
+    };
 
-    facade.initiate_communication(id, body.into()).await;
+    facade.save(swap).await;
```

Comment:
```suggestion
    facade.save(swap).await?;
```

`save` should return a result.
I think that we should proceed if we cannot save the swap in the DB as it means that if cnd is stopped later, we will not be able to resume the swap and hence the user risks to lose his assets.

Reply:
`save` actually return a result so let's use it (I guess clippy would say the same?).

Reply:
Yeah, I back and forth'ed on this point during development.  I totally agree with you but in the end I decided that adding the error return was out of scope for this ticket, since the facade's `save` implementation is empty changing the signature of an empty function seemed like a dirty thing to do.  I elected to leave the correction of the error handling for when the facade method is done (and I assume this will be within the scope of the ticket that does the save implementations in the database (introduced as `unimplemented` in this PR).  Thoughts?

Reply:
At best, you nnow ask the next developer to go through the same process than we just did to decide whether or not the failure of save should stop the action.
At worst, this will not be caught up by the next dev/reviewers and user will lose money.

> since the facade's save implementation is empty changing the signature of an empty function seemed like a dirty thing to do

I would disagree, on the contrary. Creating a skeleton with functions that have an unimplemented body is the perfect timing to update the interfaces through and through. After all, if you create a function but do not implement its logic, isn't it because you want to do just that: refine and prepare the interfaces so that the follow-up work will be about implementing the logics and not revising the interfaces again?

Reply:
> At worst, this will not be caught up by the next dev/reviewers and user will lose money.

The facade save method has to delegate to the db save trait which returns an error - I don't see how it would be possible to forget to deal with this error.